pkg/resource/graph/dependency_graph.go

```diff
@@ -69,6 +69,53 @@ func (dg *DependencyGraph) DependingOn(res *resource.State) []*resource.State {
 	return dependents
 }
 
+// DependenciesOf returns a ResourceSet of resources upon which the given resource depends.
+func (dg *DependencyGraph) DependenciesOf(res *resource.State) ResourceSet {
```

Comment:
Is there any reason not to include the parent here? In a very real sense it is a dependency--a resource R with a parent P cannot be registered prior to P.

Reply:
Nope, and it probably should - I have to explicitly add the parent to the set of dependencies in the step generator anyway for this reason.